### PR TITLE
Add event service and integrate Dashboard table with backend

### PR DESCRIPTION
**Pull request by joshuali925**
Signed-off-by: Joshua Li <joshuali925@gmail.com>

### Description
- add event routes and services
- update data modal to be consistent with backend
- integrate Dashboard UI with backend
- removed histogram until backend supports it

TODO: 
- need to test inverted filters once backend supports it, currently they will not work
- need to add histogram when backend supports it

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-notifications/public/pages/Notifications/components/SearchBar/utils/filterHelpers.tsx [hunk 1]
```diff
@@ -110,12 +112,22 @@ export const getValueComponent = (
...
+    options = Object.entries(SEVERITY_TYPE).map(([key, value]) => ({
+      label: value,
+      value: key,
+    }));
+  else if (field === 'Source')
+    options = Object.entries(NOTIFICATION_SOURCE).map(([key, value]) => ({
+      label: value,
+      value: key,
+    }));
+  else if (field === 'Status') options = [{ label: 'Sent' }, { label: 'Error' }];
```
Comment (davidcui-amzn) [inline dashboards-notifications/public/pages/Notifications/components/SearchBar/utils/filterHelpers.tsx:130]: np: should we stay consistent with formatting in these `else if` blocks?

File: dashboards-notifications/public/pages/Notifications/containers/Notifications.tsx [hunk 2]
```diff
@@ -281,33 +268,33 @@ export default class Notifications extends Component<
...
           setFilters={this.setFilters}
           refresh={this.getNotifications}
         />
 
-        {/* TODO: change back to items.length > 0 */}
-        {this.state.search.length === 0 ? (
+        {this.state.loading || this.state.total > 0 ? (
           <>
-            <EuiSpacer />
+            {/* <EuiSpacer />
```
Comment (davidcui-amzn) [inline dashboards-notifications/public/pages/Notifications/containers/Notifications.tsx:280]: Any reason we are leaving this commented-out code block?

**Review by davidcui-amzn — APPROVED** [review]

File: dashboards-notifications/public/pages/Notifications/components/SearchBar/utils/filterHelpers.tsx (hunk 1 above)
Comment (joshuali925) [reply dashboards-notifications/public/pages/Notifications/components/SearchBar/utils/filterHelpers.tsx]: updated, thanks

File: dashboards-notifications/public/pages/Notifications/containers/Notifications.tsx (hunk 2 above)
Comment (joshuali925) [reply dashboards-notifications/public/pages/Notifications/containers/Notifications.tsx:280]: This is temporary, I'll uncomment it when backend supports histogram queries

**Pull request merged**